atlasdb-cli/src/main/java/com/palantir/atlasdb/cli/command/debug/ComputeSweepStats.java

```diff
@@ -0,0 +1,95 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.cli.command.debug;
+
+import java.util.Map;
+
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.Maps;
+import com.palantir.atlasdb.cli.command.SingleBackendCommand;
+import com.palantir.atlasdb.cli.output.OutputPrinter;
+import com.palantir.atlasdb.keyvalue.api.KeyValueService;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.keyvalue.api.TargetedSweepMetadata;
+import com.palantir.atlasdb.keyvalue.api.WriteReference;
+import com.palantir.atlasdb.keyvalue.api.WriteReferencePersister;
+import com.palantir.atlasdb.schema.generated.TargetedSweepTableFactory;
+import com.palantir.atlasdb.services.AtlasDbServices;
+import com.palantir.atlasdb.sweep.queue.id.SweepTableIndices;
+
+import io.airlift.airline.Command;
+
+@Command(name = "compute-sweep-stats",
+        description = "Output some statistics about the live sweep queue.")
+public class ComputeSweepStats extends SingleBackendCommand {
+    private static final OutputPrinter printer = new OutputPrinter(LoggerFactory.getLogger(ComputeSweepStats.class));
+
+    private Map<TableReference, Long> countByTable = Maps.newHashMap();
+    private Map<Boolean, Long> countByType = Maps.newHashMapWithExpectedSize(2);
+    private Map<TableReference, Boolean> tableToType = Maps.newHashMap();
+
+    @Override
+    public int execute(AtlasDbServices services) {
+        KeyValueService kvs = services.getKeyValueService();
+        WriteReferencePersister writeReferencePersister = new WriteReferencePersister(new SweepTableIndices(kvs));
+
+        services.getTransactionManager().runTaskReadOnly(t -> {
+            TargetedSweepTableFactory.of().getSweepableCellsTable(t).getAllRowsUnordered().hintBatchSize(10_000)
+                    .forEach(rowResult -> {
+                        TargetedSweepMetadata metadata =
+                                TargetedSweepMetadata.BYTES_HYDRATOR.hydrateFromBytes(
+                                        rowResult.getRowName().getMetadata());
+
+                        rowResult.getColumnValues().stream().forEach(columnValue -> {
+                            WriteReference writeRef = writeReferencePersister.unpersist(columnValue.getValue());
+                            TableReference tableRef = writeRef.tableRef();
+                            updateStats(tableRef, metadata.conservative());
+                        });
+                    });
+            return 0;
+        });
+        printStats();
+        return 0;
+    }
+
+    private void updateStats(TableReference tableRef, boolean isConservative) {
```

Comment:
Note that dedicated rows will not have a real value for the `WriteReference` in the non-dedicated row. I've not checked what is written, though.